scene/resources/animation.cpp

```diff
@@ -5602,13 +5605,81 @@ Variant Animation::add_variant(const Variant &a, const Variant &b) {
 			return (a.operator Transform3D()) * (b.operator Transform3D());
 		}
 		default: {
+			if (a.is_array()) {
+				switch (a.get_type()) {
+					case Variant::ARRAY: {
+						Array arr_a = a;
+						Array arr_b = b;
+						int size = arr_a.size();
+						if (size == 0 || arr_b.size() != size) {
+							return a;
+						}
```

Comment:
```cpp
int size = MAX(arr_a.size(), arr_b.size());
```
Shouldn't we consider the maximum size? It seems that at least in `blend_packed_array()` the maximum size should be considered if you implement it.

Reply:
If maximum size is used, it would go out of bounds when operating on one of the two arrays when the sizes do not match If anything, it could be MIN(), and anything above that index is ignored.

Reply:
You can use initial values for elements outside of bounds. Then if resize() is properly done, access outside of bounds will not occur.

Reply:
For example, if we do the following tween:
```
[1, 2] -> [2, 4, 2] -> [2, 1]
```
The result would be as follows:

When using MIN: ```[1, 2] -> [2, 4] -> [2, 1]```
When using MAX: ```[1, 2] -> [2, 4, 2] -> [2, 1, 0]```

If you use MIN, the final result may be smaller than the element in the middle of the tween, so someone may access out of bounds unless they know after the tween is over that it is smaller than the element in the middle of the tween. So it is safer to use MAX.

Reply:
I do understand your theoretical implementation. I'm just not sure it would be expected behavior to make a tweener capable of resizing the Array it ends up returning.

Reply:
At least `interpolate_variant()` implicitly resizes the array for String.

Reply:
I tried interpolating polygon from the default color array to another array and the result was that the value jumped to the final value at the end of animation. Would using MAX handle this more gracefully?